sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala

```diff
@@ -206,6 +206,12 @@ class CSVOptions(
     sep
   }
 
+  /**
+   * option to infer date Type in the schema
```

Comment:
For me as an user, it is not clear the relation between the `inferSchema` option and this one. Does this option enable inferring independently from inferSchema`?

Reply:
This option inferDateType is added to -> This is to keep in sync with older version of spark, If someone wants to use the dateType, they can enable it in the option. This is just to prevent any migration issue to spark-3.2.0 from the older version. if they don't enable this option inferDateType, It will infer it as StringType.

Where as on other hand inferSchema  is to enable the inferring of schema. 
If inferSchema is enabled and inferDateType option is enabled in that case on reading the schema is this will infer at data type as DateType format instead of  StringType